internal/provider/serverless_branch_resource.go

```diff
@@ -0,0 +1,507 @@
+	var clusterId string
+	var branchId string
+
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root("cluster_id"), &clusterId)...)
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root("branch_id"), &branchId)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "delete serverless_branch_resource")
+	_, err := r.provider.ServerlessClient.DeleteBranch(ctx, clusterId, branchId)
+	if err != nil {
+		resp.Diagnostics.AddError("Delete Error", fmt.Sprintf("Unable to call DeleteBranch, got error: %s", err))
+		return
+	}
+}
+
+func (r serverlessBranchResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddWarning("Update Warning", "Update is not supported for serverless branch")
+	// get state
+	var state serverlessBranchResourceData
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	branch, err := r.provider.ServerlessClient.GetBranch(ctx, state.ClusterId.ValueString(), state.BranchId.ValueString(), branchV1beta1.BRANCHSERVICEGETBRANCHVIEWPARAMETER_BASIC)
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("Unable to call GetBranch, error: %s", err))
+		return
+	}
+
+	state.State = types.StringValue(string(*branch.State))
+	state.UpdateTime = types.StringValue(branch.UpdateTime.String())
+
+	// save into the Terraform state.
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r serverlessBranchResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, ",")
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		resp.Diagnostics.AddError(
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected import identifier with format: cluster_id, branch_id. Got: %q", req.ID),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("cluster_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("branch_id"), idParts[1])...)
+}
+
+func buildCreateServerlessBranchBody(data serverlessBranchResourceData) (branchV1beta1.Branch, error) {
+	displayName := data.DisplayName.ValueString()
+	parentId := data.ParentId.ValueString()
+	body := branchV1beta1.Branch{
+		DisplayName: displayName,
+		ParentId:    &parentId,
+	}
+
+	if data.ParentTimestamp.ValueString() != "" {
+		parentTimestampStr := data.ParentTimestamp.ValueString()
+		parentTimestamp, err := time.Parse(time.RFC3339, parentTimestampStr)
+		if err != nil {
+			return branchV1beta1.Branch{}, err
+		}
+		body.ParentTimestamp = *branchV1beta1.NewNullableTime(&parentTimestamp)
+	}
+
+	if data.Endpoints != nil {
+		publicEndpointsDisabled := data.Endpoints.Public.Disabled.ValueBool()
+		body.Endpoints = &branchV1beta1.BranchEndpoints{
+			Public: &branchV1beta1.BranchEndpointsPublic{
+				Disabled: &publicEndpointsDisabled,
+			},
+		}
+	}
```

Comment:
?

Reply:
I set this because it's an optional field in swagger